### PR TITLE
Modernise `IsStartNode()`

### PR DESCRIPTION
**Pull request by aubergine10**
Replaces obsolete `ExtSegmentManager.IsStartNode()` with `NetSegment` extensions - either `IsStartnode()` (always returns a bool) or `IsStartNode()` (may return `null` if node not attached to segment) as applicable.

Additionally, updates a few encountered `for` loops like `for (int i = 0; i < 8; ++i)` to more descriptive `for (int segmentIndex = 0; segmentIndex < Constants.MAX_SEGMENTS_OF_NODE; ++segmentIndex)`.

Small number of other minor tweaks - eg. to replace some other obsolete stuff in proximity/relation to the main task.

Probably has a lot of overlap with stalled PR #1272.

### CODE REVIEW

**Comment by kianzarrin** [comment]
This is kind of like the canceled #1272 (because it got too big). I have some regular expression over there if you want to do the same to other extension methods.

**Comment by kianzarrin** [comment]
we also have `IsstartNode()` that does returns a simple bool.

**Comment by aubergine10** [comment]
> we also have `IsstartNode()` that does returns a simple bool.

I assume you mean `IsStartnode()` (lowercase `n`) ? I'm already using that one where simple bool is required (it's slightly faster as it only does one comparison).

I couldn't find any `IsstartNode()` method with the lowercase `s`.

File: TLM/TLM/Util/Record/LaneConnectionRecord.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public class LaneConnectionRecord : IRecordable {
                     var laneData = new LaneConnectionRecord {
                         LaneId = laneIdAndIndex.laneId,
                         LaneIndex = (byte)laneIdAndIndex.laneIndex,
-                        StartNode = (bool)extSegmentManager.IsStartNode(segmentId, nodeId),
+                        StartNode = netSegment.IsStartnode(nodeId),
```
Comment (kvakvs) [inline TLM/TLM/Util/Record/LaneConnectionRecord.cs:106]: `NetSegment.IsStartnode` capitalisation, should be `IsStartNode`

**Review by kvakvs — APPROVED** [review]
And then I kinda like this code change

File: TLM/TLM/Util/Record/LaneConnectionRecord.cs (hunk 1 above)
Comment (aubergine10) [reply TLM/TLM/Util/Record/LaneConnectionRecord.cs:106]: The `IsStartnode` (lowercase `n`) always returns a bool so I've been using that in cases where there were no `null` checks (the `IsStartNode` - capital `N` - version of the extension will return `null` if the node does not belong to segment).

Comment (aubergine10) [reply TLM/TLM/Util/Record/LaneConnectionRecord.cs:106]: BTW, I didn't add the `IsStartnode` extension, it was already there. I moved both extensions next to each other in recent PR.

Maybe the `IsStartNode` (one that checks node belongs to segment) should be renamed `IsValidStartNode`? Then the other one can be `IsStartNode` with correct capitalisation. Or something along those lines...? If we want to rename the methods this PR is a good place to do it.

Comment (kvakvs) [reply TLM/TLM/Util/Record/LaneConnectionRecord.cs:106]: So its ok to rename 2500000 other locations in the code, but it makes people hesitant whether the bad naming should be fixed or not? 😀 of course, rename it to something better.

Comment (aubergine10) [reply TLM/TLM/Util/Record/LaneConnectionRecord.cs]: done

**Comment by aubergine10** [comment]
bump :)

**Review by krzychu124 — APPROVED** [review]
Looks good, didn't notice problems

**Pull request merged**